aws-datastore/src/main/java/com/amplifyframework/datastore/storage/StorageItemChange.java

```diff
@@ -133,12 +131,13 @@ public T item() {
      * @param recordFactory A component capable of generating records.
      * @return A Record representation of this instance.
      */
+    @NonNull
     public Record toRecord(@NonNull RecordFactory recordFactory) {
         return Objects.requireNonNull(recordFactory).toRecord(this);
     }
 
     @Override
-    public boolean equals(Object thatObject) {
+    public boolean equals(@Nullable Object thatObject) {
         if (this == thatObject) {
             return true;
```

Comment:
This looks like the default equals functionality - is there a need to override it that I'm missing here?

Reply:
The [default `equals(...)`](https://docs.oracle.com/javase/7/docs/api/java/lang/Object.html#equals(java.lang.Object)) works like this:
> for any non-null reference values x and y, this method returns true if and only if x and y refer to the same object (x == y has the value true).

So, that'd be equivalent to an implementation like `return this == thatObject`.

But here, I say, "okay, if they're the same object, yea, they're equal. BUT ALSO.... they're equal if they're fields are equal, _even if they're not_ the same object."

Reply:
Argh, the diff made it look like there was nothing more to this method than what's shown above. Ignore this comment.